src/ChatBot/Twitch/ChatMessage.php

```diff
@@ -24,13 +28,38 @@ public function __construct(string $channel, string $username, string $message)
     public static function parseIRCMessage(string $message): ?ChatMessage
     {
         try {
-            preg_match("/:(.*)\!.*#(.*) :(.*)/", $message, $matches);
+            preg_match("/:(.*)\!.*#(\S+) :(.*)/", $message, $matches);
 
             return new ChatMessage($matches[2], $matches[1], $matches[3]);
         } catch (\Exception $exception) {
-            echo $exception->getMessage();
+            echo $exception->getMessage()."\r\n";
         }
 
         return null;
     }
+
+    public function toHtml(string $emoteSize = Emote::IMAGE_SIZE_MD): string
+    {
+        $emoteList = collect($this->possibleEmotes)
```

Comment:
Wouldn't it be better to do this only once in constructor?

Reply:
currently we use "possibleEmotes" only here directly. And it could be changed from the outside all the time because is a public variable. 
So I think the safer way is to change it into a collection here